test/built-ins/BigInt/asIntN/bigint-tobigint.js

```diff
@@ -0,0 +1,26 @@
+// Copyright (C) 2017 Josh Wolfe. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+esid: pending
+description: BigInt.asIntN type coercion for bigint parameter
+info: >
+  BigInt.asIntN ( bits, bigint )
+
+  2. Let bigint ? ToBigInt(bigint).
+features: [Symbol]
+---*/
+
+function MyError() {}
+
+assert.sameValue(BigInt.asIntN(3, Object(10n)), 2n);
+assert.sameValue(BigInt.asIntN(3, {valueOf:()=>10n}), 2n);
+assert.throws(MyError, () => BigInt.asIntN(0, {valueOf(){throw new MyError();}}));
```

Comment:
> I don't think the distinctions you're raising are extremely likely to cause bugs.

That's a common phrase before tests. :)

> I thought you were opposed to too many redundant combinations of tests, as you've been removing them from the repository. This test is trying to get at some of what's involved in type conversions, but doesn't do all possible tests.

I don't think these would be redundant, but they would check other easy to observe units (getting and calling other methods). With that, we could also check the order of them being called, as valueOf throwing a custom error after a valid @@toPrimitive return, e.g.